### PR TITLE
add single result query support for reactive repositories

### PR DESCRIPTION
**Pull request by Blackbaud-EricSlater**
@Blackbaud-MikeLueders

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/ReactiveCosmosQueryExecution.java [hunk 1]
```diff
@@ -39,6 +43,31 @@ public Object execute(DocumentQuery query, Class<?> type, String container) {
...
+
+        @Override
+        public Object execute(DocumentQuery query, Class<?> type, String container) {
+            return operations.find(query, type, container)
+                .buffer(2)
+                .map((vals) -> {
+                    if (vals.size() > 1) {
+                        throw new CosmosDBAccessException("Too many results - return type "
+                            + returnedType.getReturnedType() +
+                            " is not of type Iterable but find returned multiple results");
```
Comment (Blackbaud-MikeLueders) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/ReactiveCosmosQueryExecution.java]: `find` should be `found`

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/AbstractReactiveCosmosQuery.java [hunk 2]
```diff
@@ -59,4 +64,8 @@ public ReactiveCosmosQueryMethod getQueryMethod() {
 
     protected abstract boolean isExistsQuery();
 
+    protected boolean isReactiveSingleResultQuery() {
```
Comment (Blackbaud-MikeLueders) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/AbstractReactiveCosmosQuery.java]: is this protected for a reason?

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/ReactiveCosmosQueryExecution.java (hunk 1 above)
Comment (Blackbaud-EricSlater) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/ReactiveCosmosQueryExecution.java]: changed to a more reactive appropriate error message...
"Too many results - Expected Mono<"
                            + returnedType.getReturnedType() +
                            "> but query returned multiple results"

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/AbstractReactiveCosmosQuery.java (hunk 2 above)
Comment (Blackbaud-EricSlater) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/AbstractReactiveCosmosQuery.java]: no. changed to private

**Pull request merged**